format/fit/fit.go

```diff
@@ -0,0 +1,343 @@
+package fit
+
+import (
+	"embed"
+	"fmt"
+	"sort"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/fit/mappers"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed fit.md
+var fitFS embed.FS
+
+func init() {
+	interp.RegisterFormat(
+		format.FIT,
+		&decode.Format{
+			Description: "ANT FIT",
+			Groups:      []*decode.Group{format.Probe},
+			DecodeFn:    decodeFIT,
+		})
+
+	interp.RegisterFS(fitFS)
+}
+
+var fitCRCTable = [16]uint16{
+	0x0000, 0xCC01, 0xD801, 0x1400, 0xF001, 0x3C00, 0x2800, 0xE401,
+	0xA001, 0x6C00, 0x7800, 0xB401, 0x5000, 0x9C01, 0x8801, 0x4400,
+}
+
+func calcCRC(bytes []byte) uint16 {
+	var crc uint16
+	crc = 0
+	for i := 0; i < len(bytes); i++ {
+		// compute checksum of lower four bits of byte
+		var checkByte = bytes[i]
+		var tmp = fitCRCTable[crc&0xF]
+		crc = (crc >> 4) & 0x0FFF
+		crc = crc ^ tmp ^ fitCRCTable[checkByte&0xF]
+		tmp = fitCRCTable[crc&0xF]
+		crc = (crc >> 4) & 0x0FFF
+		crc = crc ^ tmp ^ fitCRCTable[(checkByte>>4)&0xF]
+	}
+
+	return crc
+}
+
+type fitContext struct {
+	dataSize   uint64
+	headerSize uint64
+}
+
+type dataRecordContext struct {
+	compressed         bool
+	data               bool
+	localMessageType   uint64
+	hasDeveloperFields bool
+}
+
+type fileDescriptionContext struct {
+	devIdx        uint64
+	fDefNo        uint64
+	typ           string
+	name          string
+	unit          string
+	nativeFieldNo uint64
+	nativeMsgNo   uint64
+}
+
+type devFieldDefMap map[uint64]map[uint64]mappers.FieldDef
+type localFieldDefMap map[uint64]map[uint64]mappers.FieldDef
+type localMsgIsDevDef map[uint64]bool
+
+func fitDecodeFileHeader(d *decode.D, fc *fitContext) {
+	frameStart := d.Pos()
+
+	headerSize := d.FieldU8("headerSize")
```

Comment:
Would it be ok to use jq-style `snake_case` for all the field names? i've tried to use that in all other decoders even for symbolic names unless the protocol/spec etc is very specific about it

Reply:
I agree. It fits better with all the field names from the Profile.xlsx as well.